src/app/store/networks/networks.hooks.ts

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useMemo } from 'react';
+import { useAtom } from 'jotai';
+import { StacksMainnet, StacksNetwork, StacksTestnet } from '@stacks/network';
+import { ChainID } from '@stacks/transactions';
+
+import { whenChainId } from '@app/common/transactions/transaction-utils';
+import { useAppDispatch } from '@app/store';
+
+import { useCurrentNetwork } from './networks.selectors';
+import { DefaultNetworkIds, Network } from './networks.slice';
+import { networksActions } from './networks.actions';
+import { currentNetworkAtom } from './networks';
+
+// TODO: Remove - this is temporary for network refactor
+export function useSetCurrentNetworkAtom() {
+  const [_, setCurrentNetworkAtom] = useAtom(currentNetworkAtom);
+  const network = useCurrentNetwork();
+
+  useEffect(() => {
+    if (network) setCurrentNetworkAtom(network);
+  }, [network, setCurrentNetworkAtom]);
+}
+
+export function useCurrentNetworkState() {
+  const currentNetwork = useCurrentNetwork();
+
+  return useMemo(() => {
+    const isTestnet = currentNetwork?.chainId === ChainID.Testnet;
+    const mode = isTestnet ? DefaultNetworkIds.testnet : DefaultNetworkIds.mainnet;
+    return { ...currentNetwork, isTestnet, mode };
+  }, [currentNetwork]);
+}
+
+export function useCurrentStacksNetworkState(): StacksNetwork {
+  const currentNetwork = useCurrentNetwork();
+
+  return useMemo(() => {
+    if (!currentNetwork) throw new Error('No current network');
+
+    const stacksNetwork = whenChainId(currentNetwork.chainId || 1)({
+      [ChainID.Mainnet]: new StacksMainnet({ url: currentNetwork.url }),
+      [ChainID.Testnet]: new StacksTestnet({ url: currentNetwork.url }),
+    });
+
+    stacksNetwork.bnsLookupUrl = currentNetwork.url || '';
+    return stacksNetwork;
+  }, [currentNetwork]);
+}
+
+export function useNetworksActions() {
+  const dispatch = useAppDispatch();
+
+  return useMemo(
+    () => ({
+      currentNetworkChanged(id: string) {
+        return dispatch(networksActions.currentNetworkChanged(id));
+      },
+      newNetworkAdded({ chainId, id, name, url }: Network) {
+        return dispatch(networksActions.newNetworkAdded({ chainId, id, name, url }));
+      },
+      networkRemoved(id: string) {
```

Comment:
"network removed" describes the result of the action we want, not the action itself. `removeNetwork` would be a better name